pkg/invoke/invoker.go

```diff
@@ -479,6 +480,9 @@ func (i *Invoker) createRun(ctx context.Context, c kclient.WithWatch, thread *v1
 				return err
 			}
 			thread.Status.CurrentRunName = run.Name
+			if err := retention.SetLastUsedTime(ctx, c, thread); err != nil {
```

Comment:
I implemented this so that, every time we change the `CurrentRunName` on a thread, we also update its last used time. I figured this was the easiest way to determine thread usage.